redefinition of typedef ‘NIM_STATIC_ASSERT_AUX’  in x32

I try to use musl-gcc build a 32bit file。
system: centos6.8 i386
command: nim --gcc.exe:musl-gcc --gcc.linkerexe:musl-gcc --passL:-static -d:release -d:danger -d:strip --opt:size c xxx.nim
error out:
 
Hint: used config file '/opt/nim/config/nim.cfg' [Conf]
Hint: used config file '/opt/nim/config/config.nims' [Conf]
...............................................
Hint: musl-gcc -c  -w -fmax-errors=3 -Os -fno-ident   -I/opt/nim/lib -I/root -o /root/.cache/nim/xxx_r/stdlib_assertions.nim.c.o /root/.cache/nim/xxx_r/stdlib_assertions.nim.c [Exec]
In file included from /root/.cache/nim/xxx_r/stdlib_assertions.nim.c:12:
/opt/nim/lib/nimbase.h:542: error: redefinition of typedef ‘NIM_STATIC_ASSERT_AUX’
/opt/nim/lib/nimbase.h:323: note: previous declaration of ‘NIM_STATIC_ASSERT_AUX’ was here
Error: execution of an external program failed: 'musl-gcc -c  -w -fmax-errors=3 -Os -fno-ident   -I/opt/nim/lib -I/root -o /root/.cache/nim/xxx_r/stdlib_assertions.nim.c.o /root/.cache/nim/xxx_r/stdlib_assertions.nim.c'

I'm able to reproduce the issue by using gcc 4.1 or 4.2:
```
./bin/nim c --gcc.exe:gcc-4.2 koch.nim
[trimmed]
.........................................................
CC: stdlib_assertions.nim
CC: stdlib_io.nim
CC: stdlib_system.nim
CC: stdlib_parseutils.nim
CC: stdlib_math.nim
In file included from /home/fede/.cache/nim/koch_d/stdlib_assertions.nim.c:8:
/home/fede/projects/Nim/lib/nimbase.h:546: error: redefinition of typedef ‘NIM_STATIC_ASSERT_AUX’
/home/fede/projects/Nim/lib/nimbase.h:327: error: previous declaration of ‘NIM_STATIC_ASSERT_AUX’ was here
In file included from /home/fede/.cache/nim/koch_d/stdlib_io.nim.c:8:
/home/fede/projects/Nim/lib/nimbase.h:546: error: redefinition of typedef ‘NIM_STATIC_ASSERT_AUX’
/home/fede/projects/Nim/lib/nimbase.h:327: error: previous declaration of ‘NIM_STATIC_ASSERT_AUX’ was here
CC: stdlib_unicode.nim
[trimmed]
```

--- a/lib/nimbase.h
+++ b/lib/nimbase.h
@@ -272,11 +272,15 @@ __EMSCRIPTEN__
 #elif defined(__cplusplus)
 #define NIM_STATIC_ASSERT(x, msg) static_assert((x), msg)
 #else
-#define NIM_STATIC_ASSERT(x, msg) typedef int NIM_STATIC_ASSERT_AUX[(x) ? 1 : -1];
+#define _NIM_STATIC_ASSERT_FINAL(x, append_name) typedef int NIM_STATIC_ASSERT_AUX ## append_name[(x) ? 1 : -1];
+#define _NIM_STATIC_ASSERT_STAGE_3(x, line)      _NIM_STATIC_ASSERT_FINAL(x, _AT_LINE_##line)
+#define _NIM_STATIC_ASSERT_STAGE_2(x, line)      _NIM_STATIC_ASSERT_STAGE_3(x, line)
+#define NIM_STATIC_ASSERT(x, msg)                _NIM_STATIC_ASSERT_STAGE_2(x,__LINE__)
 // On failure, your C compiler will say something like:
-//   "error: 'NIM_STATIC_ASSERT_AUX' declared as an array with a negative size"
-// we could use a better fallback to also show line number, using:
-// http://www.pixelbeat.org/programming/gcc/static_assert.html
+//   "error: 'NIM_STATIC_ASSERT_AUX_AT_LINE_XXX' declared as an array with a negative size"
+// Adding the line number helps to avoid redefinitions which are not allowed in
+// old GCC versions, however the order of evaluation for __LINE__ is a little tricky,
+// hence all the helper macros. See https://stackoverflow.com/a/3385694 for more info.
 #endif
 
 /* C99 compiler? */

